### ISSUE TITLE
Build Error when adding azure functions durable extension to .net framework project

### ISSUE THREAD

**Issue by keyneom** [Azure/azure-functions-durable-extension#796]
**Describe the bug**
When upgrading from 1.8.1 to 1.8.2 I receive build errors. To verify it is not just an issue with the project I have created a brand new azure functions (v1) project (it built successfully to start) and then added the 1.8.2 nuget package. After doing so, the project fails to build with the following error:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Azure.WebJobs, Version=2.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
File name: 'Microsoft.Azure.WebJobs, Version=2.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
   at System.ModuleHandle.ResolveType(RuntimeModule module, Int32 typeToken, IntPtr* typeInstArgs, Int32 typeInstCount, IntPtr* methodInstArgs, Int32 methodInstCount, ObjectHandleOnStack type)
   at System.ModuleHandle.ResolveTypeHandleInternal(RuntimeModule module, Int32 typeToken, RuntimeTypeHandle[] typeInstantiationContext, RuntimeTypeHandle[] methodInstantiationContext)
   at System.Reflection.RuntimeModule.ResolveType(Int32 metadataToken, Type[] genericTypeArguments, Type[] genericMethodArguments)
   at System.Reflection.CustomAttribute.FilterCustomAttributeRecord(CustomAttributeRecord caRecord, MetadataImport scope, Assembly& lastAptcaOkAssembly, RuntimeModule decoratedModule, MetadataToken decoratedToken, RuntimeType attributeFilterType, Boolean mustBeInheritable, Object[] attributes, IList derivedAttributes, RuntimeType& attributeType, IRuntimeMethodInfo& ctor, Boolean& ctorHasParameters, Boolean& isVarArg)
   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeModule decoratedModule, Int32 decoratedMetadataToken, Int32 pcaCount, RuntimeType attributeFilterType, Boolean mustBeInheritable, IList derivedAttributes, Boolean isDecoratedTargetSecurityTransparent)
   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeType type, RuntimeType caType, Boolean inherit)
   at System.Attribute.InternalGetAttributeUsage(Type type)
   at System.Attribute.InternalParamGetCustomAttributes(ParameterInfo param, Type type, Boolean inherit)
   at MakeFunctionJson.ParameterInfoExtensions.GetDisabledAttribute(ParameterInfo parameterInfo)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)
   at MakeFunctionJson.MethodInfoExtensions.GetDisabled(MethodInfo method)
   at MakeFunctionJson.MethodInfoExtensions.HasUnsuportedAttributes(MethodInfo method, String& error)
   at MakeFunctionJson.FunctionJsonConverter.<GenerateFunctions>d__10.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at MakeFunctionJson.FunctionJsonConverter.TryGenerateFunctionJsons()
   at MakeFunctionJson.FunctionJsonConverter.TryRun()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

Error generating functions metadata
	{projectname}	C:\Users\{username}\.nuget\packages\microsoft.net.sdk.functions\1.0.28\build\netstandard1.0\Microsoft.NET.Sdk.Functions.Build.targets	41	
```

**Investigative information**

- Durable Functions extension version: 1.8.2
- Function App version (1.0 or 2.0): 1.0
- Programming language used: C#

**To Reproduce**

1. Create a new Azure Functions v1 project
2. Add the 1.8.2 nuget package
3. Build the project
4. Should receive an error message

**Expected behavior**
Package should install without causing build errors.

**Actual behavior**
Build error after install.

**Additional context**

- Visual Studio 2019

**Comment by cgillum** [comment]
@ConnorMcMahon could you take a look? I'm guessing this is related to our change to depend on **Microsoft.Azure.WebJobs v2.3.0** instead of 2.2.0.

**Comment by ConnorMcMahon** [comment]
@keyneom, can you please share the `<PackageReference>` section of your `.csproj` file? I can't reproduce it with a blank v1 project, so it may have to do with how this interacts with some of the other packages you have imported.

**Comment by vandersmissenc** [comment]
@ConnorMcMahon We have the following package references
```
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DurableTask" Version="1.8.2" />
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.28" />
```

I pushed a reproduction project to https://github.com/vandersmissenc/functions-build-error-repro if you want to check it out.

**Comment by keyneom** [comment]
@ConnorMcMahon I can verify that the same project referenced above fails to build on my machine as well

**Comment by ConnorMcMahon** [comment]
Thank you for the repo. 

It looks like I wasn't getting the repo due to this being in the post build job when trying to create a new function.json for a function. I don't know why it is looking for that version of `Microsoft.Azure.WebJobs`, as `Microsoft.NET.Sdk.Functions` version `1.0.28` has a dependency on `2.3.0`, not `2.2.0`. I will continue to dive into this.

**Comment by ConnorMcMahon** [comment]
This seems a duplicate of [this issue](https://github.com/Azure/azure-webjobs-sdk/issues/2069) for the ServiceBus extension. Will see if we can get a fix in the Microsoft.NET.Sdk.Functions package for this.

**Comment by ConnorMcMahon** [comment]
The core issue has now been triaged with the core Functions team. It should be fixed in the next release. I will leave this issue open until then for visibility on the workaround.

**Comment by ConnorMcMahon** [comment]
To mitigate this, for 1.8.3 I will revert Microsoft.Azure.WebJobs back to 2.2.0 until the external issue is resolved.

**Comment by cgillum** [comment]
This fix has been released as part of v1.8.3.
